src/contracts/CrossChainForwarder.sol

```diff
@@ -261,6 +279,68 @@ contract CrossChainForwarder is OwnableWithGuardian, ICrossChainForwarder {
     _disableBridgeAdapters(bridgeAdapters);
   }
 
+  /// @inheritdoc ICrossChainForwarder
+  function updateRequiredConfirmationsForReceiverChain(
+    RequiredConfirmationsByReceiverChain[] memory requiredConfirmationsByReceiverChain
+  ) external onlyOwner {
+    _updateRequiredConfirmationsForReceiverChain(requiredConfirmationsByReceiverChain);
+  }
+
+  /**
+   * @notice method to shuffle an array of forwarder configurations
+   * @param arrayToShuffle array that needs to be shuffled
+   * @return shuffled array of forwarder configurations
+   */
+  function _shuffleForwarderAdapters(
+    ChainIdBridgeConfig[] memory arrayToShuffle
+  ) internal returns (ChainIdBridgeConfig[] memory) {
+    uint256 arrayLength = arrayToShuffle.length;
+    for (uint256 i = 0; i < arrayLength; i++) {
+      uint256 j = Math.getPseudoRandom(i) % arrayLength;
+      ChainIdBridgeConfig memory arrayItem = arrayToShuffle[i];
+      arrayToShuffle[i] = arrayToShuffle[j];
+      arrayToShuffle[j] = arrayItem;
+    }
+    return arrayToShuffle;
+  }
+
+  /**
+   * @notice method to get a shuffled array of forwarder bridge adapters configurations
+   * @param destinationChainId id of the destination chain to get the adapters that communicate to it
+   * @return a shuffled array of the forwarder configurations for a destination chain
+   */
+  function _getShuffledBridgeAdaptersByChain(
+    uint256 destinationChainId
+  ) internal returns (ChainIdBridgeConfig[] memory) {
+    uint256 destinationRequiredConfirmations = _requiredConfirmationsByReceiverChain[
+      destinationChainId
+    ];
+
+    ChainIdBridgeConfig[] memory forwarderAdapters = _bridgeAdaptersByChain[destinationChainId];
+
+    // If configured required confirmations for a destination network are set to 0 or are bigger than current adapters,
+    // it will use all the adapters available. This way there would be no way of breaking forwarding communication
+    // by setting wrong configuration.
+    if (
```

Comment:
Maybe the second condition is good, for when a forwarderAdapter is removed, but the `== 0` not sure. As I commented on other parts, sounds that simply should not be allowed on the setter.
And regarding the second condition, another option (maybe more correct) could be to add to the function adding/removing adapter the logic to reduce required confirmations (on removal)

Reply:
I dont think we should tie the confirmations to removal / addition of adapters. As it is independant, as this is a property more on receiver side. So that we remove adapter, doesnt mean that there should be less confirmations.